{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: sentinelsat in c:\\users\\tom\\anaconda3\\lib\\site-packages (1.1.0)\n",
      "Requirement already satisfied: html2text in c:\\users\\tom\\anaconda3\\lib\\site-packages (from sentinelsat) (2020.1.16)\n",
      "Requirement already satisfied: geomet in c:\\users\\tom\\anaconda3\\lib\\site-packages (from sentinelsat) (0.3.0)\n",
      "Requirement already satisfied: geojson>=2 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from sentinelsat) (2.5.0)\n",
      "Requirement already satisfied: requests in c:\\users\\tom\\anaconda3\\lib\\site-packages (from sentinelsat) (2.24.0)\n",
      "Requirement already satisfied: click in c:\\users\\tom\\anaconda3\\lib\\site-packages (from sentinelsat) (7.1.2)\n",
      "Requirement already satisfied: tqdm>=4.27 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from sentinelsat) (4.50.2)\n",
      "Requirement already satisfied: six in c:\\users\\tom\\anaconda3\\lib\\site-packages (from geomet->sentinelsat) (1.16.0)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from requests->sentinelsat) (3.0.4)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from requests->sentinelsat) (1.25.11)\n",
      "Requirement already satisfied: idna<3,>=2.5 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from requests->sentinelsat) (2.10)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from requests->sentinelsat) (2021.10.8)\n",
      "Collecting rasterio\n",
      "  Using cached rasterio-1.2.10.tar.gz (2.3 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.9.tar.gz (2.3 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpedqa_4qe'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_7311424a0dac4cbc95e20615eab0169b\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/44/5e/9f19e1e6fe980b59d8da8809f2e8f81eb7f0322c71914f077edcbcd9a110/rasterio-1.2.10.tar.gz#sha256=6062456047ba6494fe18bd0da98a383b6fad5306b16cd52a22e76c59172a2b5f (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpedqa_4qe' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpngb1ml8o'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_59562cc3365940658d2c0e2dabcfc11b\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/1f/70/4e7a789f4988955e4c381de80923e184f912683bbe6fc4a3a00c91efdf59/rasterio-1.2.9.tar.gz#sha256=012a4964d8db365be4fae0af9cbeba00e683e5904d5031e8ba42ccb6040cc887 (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpngb1ml8o' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmphwdqkped'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_19b4757828a849569016b0322d6226be\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/5a/64/91fcf8482d5c380b68b4f69d5b0c75785db038b6923481b8e53234cf826f/rasterio-1.2.8.tar.gz#sha256=8196b7a71fea3c1573dd48d9ab0e78955e1d26e81848cce318c4930bd96782fe (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmphwdqkped' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpnj1va3xy'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_01d1d41c9403431f92476b51204c61e0\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/bd/14/9dae549c8425c7f4d0819eb770875f813d46e97b3d75d5bbaeed22630f6a/rasterio-1.2.7.tar.gz#sha256=f97d688fd8bdf02a2d3c79fded6ba7409a62564fa59ddb43f8069c1b2610ee4f (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpnj1va3xy' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpkpgeof6s'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_d0ffba4f911f4db58e7a358af3262a52\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/d6/61/f083df6539a3ad5ddb480dd5551d55039b9f59948c68415a9737ea62081a/rasterio-1.2.6.tar.gz#sha256=24975b97fe2fc3fd282d59640baab19de431448e1b23be6b85b68fecc1362f88 (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpkpgeof6s' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpp2o5eusr'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_3d8ecf7e76ef4be3a0015ee043b9187f\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/5c/a3/2f30c89323826cba9344b55d3355df16695dff14aa312dbddb8050bb571f/rasterio-1.2.5.tar.gz#sha256=dbf85c2c3dbdc5000cb0afc24bd7625ff696bd8698661fed23d115164333d80f (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpp2o5eusr' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmprqevh9ng'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_e380bcd2583a4c55a6991d3f9301fe82\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/64/22/a8a0d384147fcbb2ecdea0a05d7d178e8692c62659fe733bd4b3966e6e95/rasterio-1.2.4.tar.gz#sha256=31248c5ddc8f138f73b166dd93384aa72d11abd0dc27dca90a190276c7976aa3 (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmprqevh9ng' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpa2gmhu02'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_4cfbc27adfc84879bae5fcda5183c73a\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/97/85/3591f1b04fc9e9038aff7d6c5ddc816e8a7c0c8362f69b70ed6455ebcebd/rasterio-1.2.3.tar.gz#sha256=d8c345e01052b70ac3bbbe100c83def813c0ab19f7412c2c98e553d03720c1c5 (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpa2gmhu02' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp4ghqoudl'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_857c9e3332ca419eb66a07d5c656195c\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/b3/e9/f7092cd8ac97f1b2a6f15bfc084b496d40653dd188dd0945cfe1bbd67abe/rasterio-1.2.2.tar.gz#sha256=8187d01f30d4e01e93f0a434df6a4cc63dd026a18b5287c242b066cb56223424 (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp4ghqoudl' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp6kqjcrov'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_9617cff99a894d7eac582729f7e16cad\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/cf/15/79f2ed3d25b02ace4c3a07168ef4a3ff34afc86537d118fd25230df6830f/rasterio-1.2.1.tar.gz#sha256=81d4181514c67dc660d7b8cf943ad55c4fb3e0a17ac893d92628ac928cfab3d8 (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp6kqjcrov' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpzb1kozbq'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_da120a86ed794284ad7228f7cc6f8d77\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/b9/bc/3bcf9fb9139112e389a4c8b914c45dac01981db211d18dae7af5a081c781/rasterio-1.2.0.tar.gz#sha256=d1c70ae16f4048dce8292a607f016833c024c941b3eca51cd773fc494172c6eb (from https://pypi.org/simple/rasterio/) (requires-python:>=3.6). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpzb1kozbq' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp7j6zj8bw'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_5c82585822204eb19acf494e910f2504\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/73/2d/3e1bc56d75bf7c4cd5df51536992b3420197f2aa5ddd4d93e12100715dff/rasterio-1.1.8.tar.gz#sha256=f7cac7e2ecf65b4b1eb78c994c63bd429b67dc679b0bc0ecfe487d3d5bf88fd5 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp7j6zj8bw' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp32xo0joi'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_2b24341fd56d4e70bc8f8e450fe3682c\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/50/bf/35e564093d84509406c3d96e147132067ddbc591966b9b52c3aa0331f1b8/rasterio-1.1.7.tar.gz#sha256=8418c024f9bf7350d03855039d2700064532bae33a3c6ee885c8b6c8fc13415a (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp32xo0joi' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpetvb1ur5'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_9bf77c286a7f475394fda62ca4856421\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/c0/e0/a055775339c04db97a92254b08d70a796ff0cc8b4c9a23f92daa1dc96612/rasterio-1.1.6.tar.gz#sha256=c098881dfe301c5e3aa43a3171985fb365d64dc4d1c0a1d8288c7870fb240efe (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpetvb1ur5' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpn6mqi27v'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_12d0356767934891afd4ce6e791cf318\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/86/43/aae52a19a69ee30d28d0374c5f22d473ba3ba98ace4a5a5330a26590df95/rasterio-1.1.5.tar.gz#sha256=ebe75c71f9257c780615caaec8ef81fa4602702cf9290a65c213e1639284acc9 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpn6mqi27v' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpuw15q9zu'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_129168b8e875442d8ade117d9adba78a\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/3a/61/c91a2faa4e7bd1b3ec915d8cb9fb6cf16fe5b85e967c0336c9f0ec837ca0/rasterio-1.1.4.tar.gz#sha256=5d7393f3a690a91813022452022df1d5a3a2c732019f24f2bd0b051fd47a03c1 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpuw15q9zu' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpl1i758ci'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_847730bc6c5047669a6e219184155135\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/d2/8a/8a7c67e75092cea6de3347db5e9e65011def24da8e632a0dd74253733d5a/rasterio-1.1.3.tar.gz#sha256=fe2eac942ae4f8a97246c8cd3977723997817b53a9058cf1959551026c6177e7 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpl1i758ci' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpx3iwn3_e'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_938d97a2407348ab80ed6fdf72061f91\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/06/f1/dd83f3ad863a98ac543e19abd485bce1ba7e545c79a64cc186668fa49aa9/rasterio-1.1.2.tar.gz#sha256=f6df894e37eefb408e4562ab9c14efcf027b0f5f279be5ff8b2105d03f2d3c37 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpx3iwn3_e' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmploysetx6'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_a39855c2878e4f78beb74bf84fab32a5\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/9b/f1/c178b41a94ab6bfb4d64a224092d7549a5b8618ce21ae644e203f56a0efa/rasterio-1.1.1.tar.gz#sha256=7571ff5b96aeb9b1ac83d56534895c0e3d5e8678568375cf83c87111e2856fcb (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmploysetx6' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpxnellkf8'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_f62f2c44284f412c96d4352bada42098\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/1a/6d/ee8b15e0d45dd7e9e7f2559e37f91915a19155cf8403948ff2082903ee30/rasterio-1.1.0.tar.gz#sha256=77aa68860da9d9382cee0294b78809b04ad4f884ea95d1dec084f2f9bab105e8 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpxnellkf8' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp8pgvpbjj'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_3404b3dff2484c0596f62324d8622cdb\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/f7/03/1dcc14b0c77ec4c4ca4bd917527bc41fdbba49f56015d535f3d781c826aa/rasterio-1.0.28.tar.gz#sha256=9def7698e7076c8f148e8bfa0b2a3334308b2f5311abe9e40269e0bf9dfc645d (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp8pgvpbjj' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp8p6o5mjw'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_35439a3633e840199516fe593bc41835\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ac/d9/c369236ff112376a298bbd615471b6786e8974656c66e9db679f432f8932/rasterio-1.0.27.tar.gz#sha256=403d2c80cddd6395f3fe5474f43333415dc0dcf430d861de82234988d20d6147 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp8p6o5mjw' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpglhcz3qe'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_0c14119065fa4ebe9da79c6dcb09d07d\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/30/7c/c66b72339b17dfb58f2260473b575ed875216c4de8f579883ef3690b8759/rasterio-1.0.26.tar.gz#sha256=9d5246631b2295b9fb417f220d009a1954aeb5895c9e27bab71fc143f4ba5633 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmpglhcz3qe' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp9fsp49jz'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_b8e8a55e3dda4d2d99255d2d36ae7ad0\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/0c/62/91d23e91e7c2b6db1ba208c1e1ced59554a0e651087bf0a854d915f08d10/rasterio-1.0.25.tar.gz#sha256=f821efe923098bd9b1792553a9f5282f6aefdb8fa638e7d966dbcdc50b6b0333 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: 'c:\\users\\tom\\anaconda3\\python.exe' 'c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' get_requires_for_build_wheel 'C:\\Users\\Tom\\AppData\\Local\\Temp\\tmp9fsp49jz' Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Tom\\\\AppData\\\\Local\\\\Temp\\\\pip-install-egstva22\\\\rasterio_8386c30300cd4e989935e6492ecfba2a\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Tom\\\\AppData\\\\Local\\\\Temp\\\\pip-install-egstva22\\\\rasterio_8386c30300cd4e989935e6492ecfba2a\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-pip-egg-info-5e6z5u1y'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_8386c30300cd4e989935e6492ecfba2a\\\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/57/84/be5689470a73f5f8988cf44b81cb3493ebb8abbbebb69768c8fc86fccdd5/rasterio-1.0.24.tar.gz#sha256=4839479621045211f66868ec49625979693450bc2e476f23e7e8ac4804eaf452 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\tom\\anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Tom\\\\AppData\\\\Local\\\\Temp\\\\pip-install-egstva22\\\\rasterio_5aea4834e23e4c9dbafe49dd9874ec7c\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Tom\\\\AppData\\\\Local\\\\Temp\\\\pip-install-egstva22\\\\rasterio_5aea4834e23e4c9dbafe49dd9874ec7c\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-pip-egg-info-i66h74hp'\n",
      "       cwd: C:\\Users\\Tom\\AppData\\Local\\Temp\\pip-install-egstva22\\rasterio_5aea4834e23e4c9dbafe49dd9874ec7c\\\n",
      "  Complete output (2 lines):\n",
      "  INFO:root:Building on Windows requires extra options to setup.py to locate needed GDAL files. More information is available in the README.\n",
      "  ERROR: A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/46/ee/f9b4bfb55b6cf1b5562e0b16bc75ab2ca3810587753e4c9944279d75540c/rasterio-1.0.23.tar.gz#sha256=45e1a4b3c5c58a0b9f1a6bfd0aa08bdc15da57fa690a1f6f26d1a6a3d3cc7aa5 (from https://pypi.org/simple/rasterio/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "ERROR: Exception:\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\urllib3\\response.py\", line 438, in _error_catcher\n",
      "    yield\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\urllib3\\response.py\", line 519, in read\n",
      "    data = self._fp.read(amt) if not fp_closed else b\"\"\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\cachecontrol\\filewrapper.py\", line 62, in read\n",
      "    data = self.__fp.read(amt)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\http\\client.py\", line 458, in read\n",
      "    n = self.readinto(b)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\http\\client.py\", line 502, in readinto\n",
      "    n = self.fp.readinto(b)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\socket.py\", line 669, in readinto\n",
      "    return self._sock.recv_into(b)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\ssl.py\", line 1241, in recv_into\n",
      "    return self.read(nbytes, buffer)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\ssl.py\", line 1099, in read\n",
      "    return self._sslobj.read(len, buffer)\n",
      "socket.timeout: The read operation timed out\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\cli\\base_command.py\", line 164, in exc_logging_wrapper\n",
      "    status = run_func(*args)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\cli\\req_command.py\", line 205, in wrapper\n",
      "    return func(self, options, args)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\commands\\install.py\", line 338, in run\n",
      "    requirement_set = resolver.resolve(\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\resolver.py\", line 92, in resolve\n",
      "    result = self._result = resolver.resolve(\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\resolvelib\\resolvers.py\", line 482, in resolve\n",
      "    state = resolution.resolve(requirements, max_rounds=max_rounds)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\resolvelib\\resolvers.py\", line 349, in resolve\n",
      "    self._add_to_criteria(self.state.criteria, r, parent=None)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\resolvelib\\resolvers.py\", line 173, in _add_to_criteria\n",
      "    if not criterion.candidates:\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\resolvelib\\structs.py\", line 151, in __bool__\n",
      "    return bool(self._sequence)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\found_candidates.py\", line 155, in __bool__\n",
      "    return any(self)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\found_candidates.py\", line 143, in <genexpr>\n",
      "    return (c for c in iterator if id(c) not in self._incompatible_ids)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\found_candidates.py\", line 47, in _iter_built\n",
      "    candidate = func()\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\factory.py\", line 201, in _make_candidate_from_link\n",
      "    self._link_candidate_cache[link] = LinkCandidate(\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\candidates.py\", line 281, in __init__\n",
      "    super().__init__(\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\candidates.py\", line 156, in __init__\n",
      "    self.dist = self._prepare()\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\candidates.py\", line 225, in _prepare\n",
      "    dist = self._prepare_distribution()\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\resolution\\resolvelib\\candidates.py\", line 292, in _prepare_distribution\n",
      "    return preparer.prepare_linked_requirement(self._ireq, parallel_builds=True)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\operations\\prepare.py\", line 482, in prepare_linked_requirement\n",
      "    return self._prepare_linked_requirement(req, parallel_builds)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\operations\\prepare.py\", line 527, in _prepare_linked_requirement\n",
      "    local_file = unpack_url(\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\operations\\prepare.py\", line 213, in unpack_url\n",
      "    file = get_http_url(\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\operations\\prepare.py\", line 94, in get_http_url\n",
      "    from_path, content_type = download(link, temp_dir.path)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\network\\download.py\", line 145, in __call__\n",
      "    for chunk in chunks:\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\cli\\progress_bars.py\", line 144, in iter\n",
      "    for x in it:\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_internal\\network\\utils.py\", line 63, in response_chunks\n",
      "    for chunk in response.raw.stream(\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\urllib3\\response.py\", line 576, in stream\n",
      "    data = self.read(amt=amt, decode_content=decode_content)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\urllib3\\response.py\", line 541, in read\n",
      "    raise IncompleteRead(self._fp_bytes_read, self.length_remaining)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\contextlib.py\", line 131, in __exit__\n",
      "    self.gen.throw(type, value, traceback)\n",
      "  File \"c:\\users\\tom\\anaconda3\\lib\\site-packages\\pip\\_vendor\\urllib3\\response.py\", line 443, in _error_catcher\n",
      "    raise ReadTimeoutError(self._pool, None, \"Read timed out.\")\n",
      "pip._vendor.urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Using cached rasterio-1.2.8.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.7.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.6.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.5.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.4.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.3.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.2.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.1.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.2.0.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.1.8.tar.gz (2.1 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Using cached rasterio-1.1.7.tar.gz (2.1 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.1.6.tar.gz (2.1 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.1.5.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.1.4.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.1.3.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.1.2.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.1.1.tar.gz (2.2 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.1.0.tar.gz (2.1 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.0.28.tar.gz (2.1 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.0.27.tar.gz (2.1 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.0.26.tar.gz (2.1 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.0.25.tar.gz (2.1 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'error'\n",
      "  Downloading rasterio-1.0.24.tar.gz (1.9 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Downloading rasterio-1.0.23.tar.gz (1.9 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Downloading rasterio-1.0.22.tar.gz (1.9 MB)\n",
      "Requirement already satisfied: folium in c:\\users\\tom\\anaconda3\\lib\\site-packages (0.12.1.post1)\n",
      "Requirement already satisfied: branca>=0.3.0 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from folium) (0.4.2)\n",
      "Requirement already satisfied: requests in c:\\users\\tom\\anaconda3\\lib\\site-packages (from folium) (2.24.0)\n",
      "Requirement already satisfied: jinja2>=2.9 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from folium) (2.11.2)\n",
      "Requirement already satisfied: numpy in c:\\users\\tom\\anaconda3\\lib\\site-packages (from folium) (1.21.4+vanilla)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from jinja2>=2.9->folium) (1.1.1)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from requests->folium) (3.0.4)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from requests->folium) (2021.10.8)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from requests->folium) (1.25.11)\n",
      "Requirement already satisfied: idna<3,>=2.5 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from requests->folium) (2.10)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\"apt\" ­Ґ пў«пҐвбп ў­гваҐ­­Ґ© Ё«Ё ў­Ґи­Ґ©\n",
      "Є®¬ ­¤®©, ЁбЇ®«­пҐ¬®© Їа®Ја ¬¬®© Ё«Ё Ї ЄҐв­л¬ д ©«®¬.\n",
      "\"apt\" ­Ґ пў«пҐвбп ў­гваҐ­­Ґ© Ё«Ё ў­Ґи­Ґ©\n",
      "Є®¬ ­¤®©, ЁбЇ®«­пҐ¬®© Їа®Ја ¬¬®© Ё«Ё Ї ЄҐв­л¬ д ©«®¬.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting git+git://github.com/geopandas/geopandas.git"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  ERROR: Error [WinError 2] Не удается найти указанный файл while executing command git version\n",
      "ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  Cloning git://github.com/geopandas/geopandas.git to c:\\users\\tom\\appdata\\local\\temp\\pip-req-build-w6a3v9v2\n",
      "Requirement already satisfied: descartes in c:\\users\\tom\\anaconda3\\lib\\site-packages (1.1.0)\n",
      "Requirement already satisfied: matplotlib in c:\\users\\tom\\anaconda3\\lib\\site-packages (from descartes) (3.3.2)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from matplotlib->descartes) (2.4.7)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from matplotlib->descartes) (0.10.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from matplotlib->descartes) (1.3.0)\n",
      "Requirement already satisfied: certifi>=2020.06.20 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from matplotlib->descartes) (2021.10.8)\n",
      "Requirement already satisfied: python-dateutil>=2.1 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from matplotlib->descartes) (2.8.1)\n",
      "Requirement already satisfied: numpy>=1.15 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from matplotlib->descartes) (1.21.4+vanilla)\n",
      "Requirement already satisfied: pillow>=6.2.0 in c:\\users\\tom\\anaconda3\\lib\\site-packages (from matplotlib->descartes) (8.0.1)\n",
      "Requirement already satisfied: six in c:\\users\\tom\\anaconda3\\lib\\site-packages (from cycler>=0.10->matplotlib->descartes) (1.16.0)\n"
     ]
    }
   ],
   "source": [
    "!pip install sentinelsat\n",
    "!pip install rasterio\n",
    "!pip install folium\n",
    "!apt install gdal-bin python-gdal python3-gdal \n",
    "!apt install python3-rtree \n",
    "!pip install git+git://github.com/geopandas/geopandas.git\n",
    "!pip install descartes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
